Adds Android foreground location support (#978)
Baseflow/flutter-geolocator: geolocator_android/CHANGELOG.md

## 3.1.0

> - Added a new option to Android settings that will allow us to configure the foreground service notification settings
> - Added wake lock and WiFI lock capabilities to the Android service
> - Added an Android service to retrieve location updates. This allows us to make the service a foreground service which will stay alive and continue processing location updates if if the application is moved into the background.

Reviewer: Nit: We should convert this into one bullet point as the current bullet points describe the implementation and release notes normally don't list implementation detail. I think we could convert this into something like:

```suggestion
- Adds support to make a foreground service and continue processing location updates when the application is moved into the background.
```

## 1.0.0

- Initial open source release.

> 

Reviewer: Nit: Please make sure all files end with an empty line. This is to ensure (older) POSIX
